controllers/argocd/util.go

```diff
@@ -1602,3 +1602,7 @@ func getApplicationSetHTTPServerHost(cr *argoproj.ArgoCD) string {
 	}
 	return host
 }
+
+func IsComponentEnabled(enabled *bool) bool {
```

Comment:
Can we have separate functions for each component ? Like `IsRedisEnabled` `IsRepoServerEnabled` `IsApplicationSetEnabled` which can take in a `Spec` object and return `true` or `false` based on the `enabled` flag value for the respective component.  The current function adds little value, as the caller of the function can directly check if the boolean value (`enabled`) is `true` or `false`

Reply:
Sure. I kept the function generic so that it can be used generically for every component as the check remains same.